### PR TITLE
Fix functional tests for legacy discover plugin

### LINKED ISSUE #4653: [Data Explorer] Fix functional tests failures for legacy discover
**Describe the bug**
Fix functional tests for legacy discover plugin.
The latest PR on the data explorer feature branch failed to pass cypress and has 6 failed functional test group.

### PR DESCRIPTION
**Pull request by abbyhu2000**
### Description

Make sure all the original functional tests pass for the legacy discover plugin.

### Issues Resolved
#4653 

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by abbyhu2000** [comment]
Failure in cigroup1 related to this bug: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4703
Failure in dashboard_state in cigroup4 related to this bug: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4698. The failed three tests have been skipped so it won't affect subsequent test outcome.

File: test/functional/apps/dashboard/dashboard_state.js [hunk 1]
```diff
@@ -88,6 +95,8 @@ export default function ({ getService, getPageObjects }) {
       expect(colorChoiceRetained).to.be(true);
     });
 
+    // the following three tests are skipped because of save search save window bug:
```
Comment (ashwin-pc) [inline test/functional/apps/dashboard/dashboard_state.js:98]: I see the comment but not the skip. Did you leave this in here accidentally?

File: test/functional/apps/dashboard/panel_context_menu.ts [hunk 2]
```diff
@@ -114,18 +117,29 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
...
+        });
+        await browser.refresh();
         await PageObjects.header.clickDiscover();
         await PageObjects.discover.clickNewSearchButton();
         await dashboardVisualizations.createSavedSearch({ name: searchName, fields: ['bytes'] });
         await PageObjects.header.waitUntilLoadingHasFinished();
         await PageObjects.header.clickDashboard();
+        // Have to add steps to actually click on the dashboard; since added browser.refresh() will make
+        // clickDashboard() to only land on the dashboard listing page
+        // We need to add browser.refresh() so clickDiscover() lands correctly on the legacy discover page
```
Comment (ashwin-pc) [inline test/functional/apps/dashboard/panel_context_menu.ts:140]: Not for right now but when we remove Legacy discover in 2.11, is it easy enough to migrate these to Discover 2.0?

**Review by ashwin-pc — APPROVED** [review]
Great job fixing these!

**Pull request merged**